src/dotnet/CarbonAware.WebApi.Tests/unitTests/SciScoreControllerTests.cs

```diff
@@ -0,0 +1,136 @@
+    }
+
+    /// <summary>
+    /// Tests that invalid time inputs respond with a badRequest error
+    /// </summary> location 
+    [Test]
+    public async Task InvalidTimeIntervalReturnsBadRequest_MarginalCarbonIntensity()
+    {
+        // Arrange
+        var data = 0.7;
+        var controller = new SciScoreController(this.MockSciScoreLogger.Object, CreateSciScoreAggregator(data).Object);
+
+        LocationInput locationInput = new LocationInput()
+        {
+            LocationType = LocationType.Geoposition.ToString()
+        };
+
+        string timeInterval = "";
+
+        SciScoreInput input = new SciScoreInput()
+        {
+            Location = locationInput,
+            TimeInterval = timeInterval
+        };
+
+        // Act
+        var carbonIntensityOutput = (await controller.GetCarbonIntensityAsync(input)) as ObjectResult;
+
+        // Assert
+        var expected = new CarbonAwareWebApiError() { Message = "timeInterval is required" };
+        TestHelpers.AssertStatusCode(carbonIntensityOutput, HttpStatusCode.BadRequest);
+        Assert.AreEqual(expected, carbonIntensityOutput.Value);
+    }
+
+    /// <summary>
+    /// Tests that invalid location inputs respond with a badRequest error
+    /// </summary> location 
+    [Test]
+    public async Task NullLocationReturnsBadRequest_MarginalCarbonIntensity()
+    {
+        // Arrange
+        var data = 0.7;
+        var controller = new SciScoreController(this.MockSciScoreLogger.Object, CreateSciScoreAggregator(data).Object);
+
+        string timeInterval = "2007-03-01T13:00:00Z/2007-03-01T15:30:00Z";
+
+        SciScoreInput input = new SciScoreInput()
+        {
+            TimeInterval = timeInterval
+        };
+
+        // Act
+        var carbonIntensityOutput = (await controller.GetCarbonIntensityAsync(input)) as ObjectResult;
+
+        // Assert
+        var expected = new CarbonAwareWebApiError() { Message = "location is required" };
+        TestHelpers.AssertStatusCode(carbonIntensityOutput, HttpStatusCode.BadRequest);
+        Assert.AreEqual(expected, carbonIntensityOutput.Value);
+    }
+
+    /// <summary>
+    /// Tests that invalid locationType inputs respond with a badRequest error
+    /// </summary> location 
+    public async Task InvalidLocationTypeReturnsBadRequest_MarginalCarbonIntensity()
+    {
+        // Arrange
+        var data = 0.7;
+        var controller = new SciScoreController(this.MockSciScoreLogger.Object, CreateSciScoreAggregator(data).Object);
+
+        LocationInput locationInput = new LocationInput()
+        {
+            LocationType = "InvalidType"
+        };
+
+        string timeInterval = "";
+
+        SciScoreInput input = new SciScoreInput()
+        {
+            Location = locationInput,
+            TimeInterval = timeInterval
+        };
+
+        // Act
+        var carbonIntensityOutput = (await controller.GetCarbonIntensityAsync(input)) as ObjectResult;
```

Comment:
Same as above, check this to make sure you can avoid the static analysis checker warning